src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -339,10 +340,18 @@ public void assertHasSameSizeAs(AssertionInfo info, Iterable<?> actual, Iterable
    * @throws AssertionError if the given {@code Iterable} does not contain the given values.
    */
   public void assertContains(AssertionInfo info, Iterable<?> actual, Object[] values) {
-    final List<?> actualAsList = newArrayList(actual);
-    if (commonCheckThatIterableAssertionSucceeds(info, actualAsList, values)) return;
+    final Collection<?> actualAsCollection = ensureActualCanBeReadMultipleTimes(actual);
+    if (commonCheckThatIterableAssertionSucceeds(info, actualAsCollection, values)) return;
     // check for elements in values that are missing in actual.
-    assertIterableContainsGivenValues(actual.getClass(), actualAsList, values, info);
+    assertIterableContainsGivenValues(actual.getClass(), actualAsCollection, values, info);
+  }
+
+  private Collection<?> ensureActualCanBeReadMultipleTimes(Iterable<?> actual) {
+    if (actual instanceof Collection<?>) {
+      return (Collection<?>) actual; // Avoid copy when actual is already a Collection.
+    } else {
+      return newArrayList(actual); //ensure the assertion works for non-repeatable iterables
+    }
```

Comment:
the else block is not necessary since we return in the if block.
could use the ? ternary operator too.